### PR TITLE
Closes #695 - create new stripe webhook for trial ending

### LINKED ISSUE #695: Add email for `stripe.customer.subscription.trial_will_end`
It would be great for Pay to send emails to remind customers when trials are ending.

> **customer.subscription.trial_will_end**
> Occurs three days before a subscription’s trial period is scheduled to end, or when a trial is ended immediately (using trial_end=now).

We might want two separate email contents for this so that we can denote the trial has ended or will end in X days.

### PR DESCRIPTION
**Pull request by cjilbert504**
Stripe will fire a `trial_will_end` event when a trial subscription is either 3 days from ending or ended immediately. This PR adds a new webhook that listens for that event and then routes to the correct mailer depending on the trial status e.g. is it ending in a few days or has it ended already?

### CODE REVIEW

File: lib/pay.rb [hunk 1]
```diff
@@ -61,6 +61,7 @@ module Billable
   @@emails.subscription_renewing = ->(pay_subscription, price) {
     (price&.type == "recurring") && (price.recurring&.interval == "year")
   }
+  @@emails.subscription_trial_will_end = true
```
Comment (excid3) [inline lib/pay.rb:64]: Do we need another setting for `subscription_trial_ended`?

File: app/mailers/pay/user_mailer.rb [hunk 2]
```diff
@@ -20,6 +20,14 @@ def payment_action_required
       mail to: to
     end
 
+    def subscription_trial_will_end
+      mail to: to
+    end
+
+    def subscription_trial_ended
+      mail to: to
```
Comment (excid3) [inline app/mailers/pay/user_mailer.rb:28]: If you get a chance, merge master into this branch and swap this to use the new `mail_arguments`

File: lib/pay.rb (hunk 1 above)
Comment (cjilbert504) [reply lib/pay.rb:64]: Oh yeah good call, probably so.

File: node_modules/.yarn-integrity [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "systemParams": "darwin-x64-93",
+  "modulesFolders": [],
+  "flags": [],
+  "linkedModules": [],
+  "topLevelPatterns": [],
+  "lockfileEntries": {},
+  "files": [],
+  "artifacts": {}
+}
```
Comment (excid3) [inline node_modules/.yarn-integrity:10]: You might want to remove this from the commit. I don't think you meant it to be in here.

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+# yarn lockfile v1
```
Comment (excid3) [inline yarn.lock:2]: Same with this, I don't think you meant to include the yarn.lock. 👍

**Pull request closed without merging**